### DISCUSSION TITLE
Completion / failed hooks

### DISCUSSION THREAD

**Discussion by RealOrangeOne** [RealOrangeOne/django-tasks#11] [Ideas]
When a task fails (or succeeds), should another task be automatically executed, creating a form of pipeline? Currently, this must be handled manually.

**Comment by RealOrangeOne** [comment]
A pre-execution hook might be useful too, as opposed to simply on success/fail.

Perhaps there should be a global pre-execution hook, too?

**Reply by mgax** [reply]
This sounds an awful lot like Django signals :)

**Reply by RealOrangeOne** [reply]
Signals with a filtered `sender` argument could be a simple way to handle this :thinking:

**Comment by jribbens** [comment]
This seems overly complicated. If you wanted function A to be followed by function B in a task, why would you not just make function A call function B? Or indeed make a task that did `A(); B()` or `try: A(); except: B()`?
